waltz-model/src/main/java/com/khartec/waltz/model/EntityReference.java

```diff
@@ -73,5 +74,14 @@ public static EntityReference mkRef(EntityKind kind, long id, String name, Strin
     }
 
 
+    public static EntityReference mkRef(Map map) {
+        return EntityReference.mkRef(
+                Enum.valueOf(EntityKind.class, map.get("kind").toString()),
+                Long.parseLong(map.get("id").toString()),
```

Comment:
this will NPE if the map doesn't conform to the expected signature.  Perhaps we need some pre-flight checks ?

Reply:
checks added